third-culture-software/bhima
# Add server-side function to nicely show exchange rate in reports

Added `exchangeRate` handlebars partial function to simplify nicely showing the current exchange rate and a warning message if it is not the enterprise currency.

Updated several reports to the new functions:
   - [ ] Cost Centers > Reports > Cost Centers Income and Expense
   - [ ] Finance > Reports > Annual Clients Report
   - [ ] Finance > Reports > Monthly Balance (monthly_balance)
   - [ ] Finance > Reports > Open Debtors
   - [ ] Finance > Reports > Unpaid Invoices Report *
   - [ ] Human Resources > Reports > Employee Standing Report (all employees)
   - [ ] Human Resources > Reports > Employee Standing Report (single employee)
   - [ ] Stock > Reports > Average Medication Costs per Patient
   - [ ] Stock > Reports > Lost Stock Report (lost_stock_report)
   - [ ] Stock > Reports > Stock Entries Report
   - [ ] Stock > Reports > Stock Exits Report
   - [ ] Stock > Reports > Stock Expiration Report
   - [ ] Stock > Reports > Stock Sheet
   - [ ] Stock > Reports > Stock Value Report (stock_value)

**Note:** You can use the check-mark boxes above to keep track on which ones you have tested! Click on one and see!

Other reports were not updated for various reasons, such as they already had related functionality.

This PR also makes cosmetic formatting changes to some of the reports above.

Closes https://github.com/IMA-WorldHealth/bhima/issues/6248

**TESTING**
- Use bhima_test and then a production dataset that uses Fc as the enterprise currency (eg, Vanga)
- For each dataset, try each of the reports above with the enterprise currency and with a non-enterprise currency.
- Verify that the exchange rate is correct and "nice" in each case.
